src/app.ts

```diff
@@ -0,0 +1,61 @@
+import { MongoDriverFactory } from "./MongoConnectorFactory";
+import { MongoDriver } from "./MongoConnector";
+import * as body from "body-parser";
+import express from "express";
+import { Console } from "console";
+
+const PORT = 3000;
+const app = express();
+
+app.use(body.json());
+
+let taskDataStore;
+MongoDriverFactory.build()
+  .then(async (datastore) => {
+    taskDataStore = datastore;
+  })
+  .catch(e => {
+    throw e;
+  });
+
+  //Gets All Tasks
+app.get("/", (req,res)=> {
+  taskDataStore.listTasks().then((tasks) => {
+    res.send(tasks);
+    console.log(tasks);
+  });
+})
+
+  //Gets specific Tasks via Id
+app.get("/task/:taskId", async (req,res)=> {
+  const task: string = await taskDataStore.readTask(req.params.taskId);
+  res.send(task);
+  console.log(task);
+});
+
+  //Posts req from body
+  //This Works 
+app.post("/", function(req,res){
+  taskDataStore.createTask(req.body).then((task) => {
+    res.send(task);
+    console.log(task +" New Task Was Created!");
+  });
+});
+
+  //Updates Information 
+app.patch("/", function(req,res){
+  taskDataStore.updateTask(req.body).then(()=> {
+    console.log("A Task has been updated");
+    res.sendStatus(200);
+  });
+});
+
+  //Deletes a Task
+app.delete("/delete/:taskId", async function(req,res){
+  await taskDataStore.deleteTask(req.params.taskId)
+  res.sendStatus(200);
+});
```

Comment:
It's great that you are using status codes!  I wanted to ask how you would design and implement a few parameter validation things (I'm not asking you to implement it, just how you would design and go about implementing):

How would you handle the taskId not being an ObjectID?  Similar to what happened when we met and the delete route wasn't working for some reason but magically worked itself out.  A 400 should be sent back if the id isn't valid (like how you send back the 200).  This may be helpful: https://mongodb.github.io/node-mongodb-native/api-bson-generated/objectid.html

How would you handle if the taskId is a valid ObjectID but the task does not exist in the database?  This should send back a 404.  It may be helpful to use an already existing function in the database connection class.

It, also, may be helpful to use what Paige suggested by separating this part out since it is business logic.

Reply:
Would the most appropriate way be to use conditionals testing isValid()?
if( input is valid)
   add to db
else
   throw appropriate status code

In regards to status, would be to have a separate status router similar to the one you demoed at SEC.

Reply:
I noticed in the documentation there is ObjectID.isValid() which returns boolean.  I know we used that when trouble-shooting the deleteTask().

Reply:
You're right about how to check for ObjectIDs being valid but I'm still a little unclear on how you would do this for not found tasks.  You wouldn't make a new route handler for this, it should be in the same logic as checking for the valid id and accessing the database to find your task.